TROUBLESHOOTING.md

```diff
@@ -1,34 +1,102 @@
+
+The Azure MCP Server can run in multiple modes. Review your MCP configuration to ensure it matches your expectations:
+
+- `azmcp server start` - Launches an MCP server with all tools enabled
+- `azmcp server start --service <service-name>` - Launches an MCP server with tools for the specified service (e.g., `storage`, `keyvault`)
+- `azmcp server start --service azure` - Launches an MCP server with a single `azure` tool that performs internal dynamic proxy and tool selection
+
+## Tool Limitations
+
+### 128-Tool Limit Issue
+
+#### Problem
 > [!WARNING]
 > Known Issue: "You may not include more than 128 tools in your request"
 
-When configuring Azure MCP with 'all' toolsets for convenience, you may encounter this error:
+When configuring Azure MCP with 'all' toolsets, you may encounter this error:
 
-![128 tools limit error](docs/images/128-tools-limit-error.png)
+![128 tools limit error](https://github.com/Azure/azure-mcp/blob/main/docs/images/128-tools-limit-error.png)
 
-### Root Cause
-  VS Code Copilot has a limitation of 128 tools maximum per request. When you combine multiple comprehensive toolsets (like GitHub MCP 'all' + Azure MCP 'all'), the total number of available tools exceeds this limit.  The solution to this problem is discussed in the VS Code repo [here](https://github.com/microsoft/vscode/issues/248021).
-### Workarounds
+#### Root Cause
+VS Code Copilot has a 128-tool limit per request. Combining multiple comprehensive toolsets (like GitHub MCP 'all' + Azure MCP 'all') exceeds this limit. See the [VS Code discussion](https://github.com/microsoft/vscode/issues/248021) for more details.
+
+#### Workarounds
 
 **Option 1: Use VS Code Custom Chat Modes (Recommended)**
-VS Code stable now supports [custom chat modes](https://code.visualstudio.com/docs/copilot/chat/chat-modes#_custom-chat-modes) that allow you to create scenario-specific tool configurations and quickly switch between them:
+VS Code supports [custom chat modes](https://code.visualstudio.com/docs/copilot/chat/chat-modes#_custom-chat-modes) for scenario-specific tool configurations:
 
 *Setup Steps:*
 1. Install your desired MCP servers (Azure, GitHub, etc.)
 2. Create custom chat modes in VS Code for different workflows
 3. Switch between modes based on your current task
-4. Use the tool picker to discover and add additional tools as needed
+4. Use the tool picker to discover and add tools as needed
 5. Stay within the 128-tool limit per mode while maintaining flexibility
 
-> [!NOTE]
-> This approach works well once VS Code resolves some current bugs where certain tools don't appear in the picker. This is an evolving feature that provides the best user experience.
-
-**Option 2: Use Selective Tool Loading (Manual Configuration)**
-Instead of loading all tools, configure targeted MCP servers for your specific needs:
+**Option 2: Use Selective Tool Loading**
+Configure targeted MCP servers for specific needs instead of loading all tools:
```

Comment:
nit; Line break
```suggestion
**Option 2: Use Selective Tool Loading**

Configure targeted MCP servers for specific needs instead of loading all tools:
```